docs/DeveloperGuide.md

```diff
@@ -1,34 +1,216 @@
 # Developer Guide
+- [Design](#1-Design)
+- [Implementation](#2-implementation)
+  - [[Proposed] Subject Feature](#21-proposed-subject-feature)
+- [Appendix A: Product Scope](#appendix-a-product-scope)
+- [Appendix B: User Stories](#appendix-b--user-stories)
+- [Appendix C: Non-Functional Requirements](#appendix-c-non-functional-requirements)
+- [Appendix D: Glossary](#appendix-d-glossary)
+- [Appendix E: Instructions for Manual Testing](#appendix-e-instructions-for-manual-testing)
 
-## Design & Implementation
+## 1. Design
+### Architecture
+The Duke class is the main class of the product. It is responsible for:
+* At app launch: Initializes the components in the correct sequence, and connects them up with each other.
+* At shut down: Shuts down the components and invokes storage method where necessary.	
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+The product also contains the following components:
+* Cards: Holds the data in the type of Card and relative operations
+* Subjects: Holds the data in the type of Subject and relative operations
+* Score: Holds the data in the type of Score and relative operations
+* Commands: Deals with user input and communicate CLI to relative methods
+* Exceptions: Deals with illegal inputs
+* Parser: Convert CLI inputs into command keywords
 
+In these components, cards and subjects have similar structure. Both of them contains a Card/Subject class and CardList/SubjectList class.
+Duke, along with all command class and Parser form the logic box of the product.
+<br />![](images/logicuml.jpg)
 
-## Product Scope
-### Target user profile
+<br />Logic box interacts with Model box, i.e. Card, CardList, Subject, SubjectList, ScoreList.
+<br />![](images/modeluml.jpg)
 
-{Describe the target user profile}
+<br />Finally the Storage box, i.e. Storage class will handle reading and writing the content to files.
+The Storage component saves the SubjectList objects in Serializable format and loads it back.
+<br />![](images/storageuml.jpg)
 
-### Value proposition
+## 2. Implementation
+### 2.1. [Proposed] Subject Feature
+#### 2.1.1. Proposed Implementation
+The subject feature is an extension to the existing flashcard feature which allows users to categorise their
```

Comment:
Just a thought, but I was trying to find where the flashcard feature is in your DG as you mentioned that it extends from it. You should include it in your DG to avoid confusion, but I can understand that you may not have time to add yet.